packages/embark-ui/src/services/web3.js

```diff
@@ -21,13 +21,12 @@ export function estimateGas({web3, contract, args}) {
 
 export function deploy({web3, contract, args}) {
   return new Promise((resolve, reject) => {
-    const cleanup = () => { promiEvent.removeAllListeners(); };
-    const promiEvent = new web3.eth.Contract(contract.abiDefinition)
+    new web3.eth.Contract(contract.abiDefinition)
       .deploy({data: `0x${contract.code}`, arguments: args})
       .send({from: web3.eth.defaultAccount})
       .on('error', reject)
       .on('receipt', resolve)
```

Comment:
```suggestion
      .once('receipt', resolve)
```

Reply:
Small thing: since promises either reject/resolve only once, I think it's better to use `.once` event handlers instead of `.on`.

Reply:
See: https://github.com/embark-framework/embark/pull/1360#discussion_r261301208.